elide-core/src/main/java/com/yahoo/elide/graphql/GraphQLEndpoint.java

```diff
@@ -73,23 +77,49 @@ public GraphQLEndpoint(
     public Response post(
             @Context SecurityContext securityContext,
             String graphQLDocument) {
-        String path = "/";
         boolean isVerbose = false;
         try (DataStoreTransaction tx = elide.getDataStore().beginTransaction()) {
             ObjectMapper mapper = elide.getMapper().getObjectMapper();
             JsonNode jsonDocument = mapper.readTree(graphQLDocument);
             final User user = tx.accessUser(getUser.apply(securityContext));
-            RequestScope requestScope = new RequestScope(path, null, tx, user, null, elide.getElideSettings());
-            ExecutionResult result = api.execute(
-                    jsonDocument.get("query").asText(),
-                    jsonDocument.get("operationName").asText(),
-                    requestScope,
-                    mapper.convertValue(jsonDocument.get("variables"), Map.class));
+            GraphQLRequestScope requestScope = new GraphQLRequestScope(tx, user, null, elide.getElideSettings());
+
+            if (!jsonDocument.has(QUERY)) {
+                return Response.status(400).entity("A `query` key is required.").build();
+            }
+
+            String query = jsonDocument.get(QUERY).asText();
+
+            String operationName = null;
+            if (jsonDocument.has(OPERATION_NAME)) {
+                operationName = jsonDocument.get(OPERATION_NAME).asText();
+            }
+
+            ExecutionResult result;
+            if (jsonDocument.has(VARIABLES)) {
```

Comment:
Do we have tests for these new options?

Reply:
These aren't new. They existed before, but were not always required. The GraphQL endpoint update is being done in a different PR and can add tests there.

Reply:
👍